Dockerfile.devcontainer

```diff
@@ -47,3 +47,17 @@ RUN \
         poetry==1.3.1 \
         && \
     true
+
+RUN \
+    mkdir -p \
+        /srv/workspace/.venv \
+        /srv/workspace/.tox \
+        /srv/workspace/.mypy_cache \
+        /srv/workspace/.pytest_cache
+
+RUN \
+    chown -R 1000:1000 \
+        /srv/workspace/.venv \
+        /srv/workspace/.tox \
+        /srv/workspace/.mypy_cache \
+        /srv/workspace/.pytest_cache
```

Comment:
Under normal operation this should be mounted from outside the docker container - but I will check tonight what is happening and ensure it works as well as it can with rootful (i.e. non-rootless) docker

Reply:
This may be relevant https://github.com/docker/compose/issues/3270

Reply:
Is it working correctly for you on the main branch?

Reply:
@edmondchuc I think I'm just trying to be too clever with what I did here.

The reasoning for adding these volumes were so that usage of the devcontainer does not contaminate your host environment while still keeping state outside of the container. This works with rootless docker on Linux, but I'm not sure how it works on MacOS and Windows - so I think best is maybe just to not try and be too clever. if it works for you with the following lines removed then we should maybe just remove them instead:

https://github.com/RDFLib/rdflib/blob/9b778b39a7759a9c94d5437e1d5f02196bdc9c51/docker-compose.yml#L12-L15

```yaml
      - dot-venv:/srv/workspace/.venv
      - dot-tox:/srv/workspace/.tox
      - dot-mypy-cache:/srv/workspace/.mypy_cache
      - dot-pytest-cache:/srv/workspace/.pytest_cache
```

Reply:
I recall trying with Docker Desktop on Windows with WSL2, macOS and also on an Ubuntu VM. I also get the same issue when I run it in GitHub Codespaces.

I agree, I think for running a dev container using Visual Studio Code, it's not necessary to have the docker compose volumes. 

Do you need the volume mounts because you're running one-off commands in the containers? E.g. `docker-compose run --rm devcontainer task validate`? 

Maybe we can keep the current `docker-compose.yml` so that you can continue running one-off commands and we can have another docker-compose file for the Visual Studio Code dev container?

https://github.com/RDFLib/rdflib/blob/9b778b39a7759a9c94d5437e1d5f02196bdc9c51/.devcontainer.json#L2

Reply:
> Do you need the volume mounts because you're running one-off commands in the containers? E.g. `docker-compose run --rm devcontainer task validate`?

The default behaviour will actually work fine, the only problem is that the venv created by the devcontainer may not be to the liking of the host system, and vice versa.

If I do `task configure` and `task validate` on my host system, it will create a .venv (I run with `POETRY_VIRTUALENVS_IN_PROJECT=1`), and pytest, mypy and other caches in my source directory. If I then run the same commands in the devcontainer with vscode or with docker compose, the .venv and cache folders from my host system will be visible in the container.

This would be fine in some cases, but in others poetry just does not like what is happening, the venvs seem to be somewhat platform specific.

But either way, we don't need it.

> Maybe we can keep the current docker-compose.yml so that you can continue running one-off commands and we can have another docker-compose file for the Visual Studio Code dev container?

We can actually just have a separate service in the same `docker-compose.yml` file as we can specify which service to run for devcontainer, which may be a bit more terse.

Reply:
looks good to me :+1: